osu.Game/Screens/Play/Player.cs

```diff
@@ -583,6 +594,8 @@ void completeResume()
             {
                 GameplayClockContainer.Start();
                 IsResuming = false;
+
+                samplePause?.Stop();
```

Comment:
By doing this in the callback of `RequestResume`, you're only making it stop playing after the user completes the resume sequence (in osu! - clicking the orange cursor). This should likely be done outside.